App/src/services/metrics.ts

```diff
@@ -0,0 +1,66 @@
+//This file is used to define metrics related types and apis
+import axios from "axios";
+import { DossierDTO } from "../models/dossier";
+import { UserDTO } from "../models/user";
+
+//types
+export interface DossierViewCountDTO {
+    guid: string;
+    dossier: DossierDTO;
+    count: number;
+}
+
+export interface DossierViewResponseDTO {
+    result: DossierViewCountDTO[];
+    nextIndex: number;
+}
+
+export interface HttpEndpointCountDTO {
+    controller: string;
+    endpoint: string;
+    fullPath: string;
+    count: number;
+}
+
+export interface HttpEndpointResponseDTO {
+    result: HttpEndpointCountDTO[];
+    nextIndex: number;
+}
+
+export interface HttpStatusCountDTO {
+    count: number;
+    httpStatus: number;
+}
+
+export interface HttpStatusResponseDTO {
+    result: HttpStatusCountDTO[];
+    nextIndex: number;
+}
+
+export interface UserDossierViewedCountDTO {
+    guid: string;
+    user: UserDTO;
+    count: number;
+}
+
+export interface UserDossierViewedResponseDTO {
+    result: UserDossierViewedCountDTO[];
+    nextIndex: number;
+}
+
+//api calls
+export function GetTopHttpStatusCodes(fromIndex: number) {
+    return axios.get("/Metrics/GetTopHttpStatusCodes?fromIndex=" + fromIndex + "&startDate=");
+}
+
+export function GetTopHitHttpEndpoints(fromIndex: number) {
+    return axios.get("/Metrics/GetTopHitHttpEndpoints?fromIndex=" + fromIndex);
+}
+
+export function GetTopViewedDossiers(fromIndex: number) {
+    return axios.get("/Metrics/GetTopViewedDossier?fromIndex=" + fromIndex);
+}
+
+export function GetTopDossierViewingUser(fromIndex: number) {
+    return axios.get("/Metrics/GetTopDossierViewingUser?fromIndex=" + fromIndex);
+}
```

Comment:
You are not using the `startDate`, you should allow the user to pass the date (without time) to have the metrics be from that date without using the default.

If you want to have the queries be HTTP encoded if needed, you can pass it as an object instead of string interpolation. However, this is nitpicking since you are passing only numbers.